src/core/compiler.jl

```diff
@@ -72,6 +75,8 @@ end
 
 macro ~(left, right)
```

Comment:
Can we store all variables in `~` and completely remove `predict` in the future?